enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/consensus/RaftMachine.java

```diff
@@ -237,27 +238,21 @@ else if ( outcome.isSteppingDown() )
         }
     }
 
-    private boolean leaderChanged( Outcome outcome, MemberId oldLeader )
+    private boolean leaderMayHaveChanged( Outcome outcome, MemberId oldLeader )
     {
-        if ( oldLeader == null && outcome.getLeader() != null )
-        {
-            return true;
-        }
-        else if ( oldLeader != null && !oldLeader.equals( outcome.getLeader() ) )
-        {
-            return true;
-        }
+        Optional<MemberId> oldLeaderOpt = Optional.ofNullable( oldLeader );
+        Optional<MemberId> outcomeLeaderOpt = Optional.ofNullable( outcome.getLeader() );
 
-        //TODO: Add logic for handling leader step-down with no replacement
+        boolean neitherPresent = !oldLeaderOpt.isPresent() && !outcomeLeaderOpt.isPresent();
```

Comment:
Is the change of logic intentional? Both being null will now return true, before false. Which will cause more notifyLeaderChanges to be called.